lib/go/k8s/configmaps.go

```diff
@@ -29,28 +29,28 @@ func ApplyConfigMap(
 					Labels:      labels,
 					Annotations: annotations,
 				},
-				Data: cmData,
+				Data: data,
 			}, v1.CreateOptions{})
 		return err
-	} else {
-		if cmData != nil && len(cmData) > 0 {
-			cm.Data = cmData
-		}
-		if cm.Labels == nil {
-			cm.Labels = make(map[string]string)
-		}
-		for k, v := range labels {
-			cm.Labels[k] = v
-		}
-		if cm.Annotations == nil {
-			cm.Annotations = make(map[string]string)
-		}
-		for k, v := range annotations {
-			cm.Annotations[k] = v
-		}
-		_, err := clientset.CoreV1().ConfigMaps(namespace).Update(context.TODO(), cm, v1.UpdateOptions{})
-		return err
 	}
+
+	if data != nil && len(data) > 0 {
+		cm.Data = data
+	}
+	if cm.Labels == nil {
+		cm.Labels = make(map[string]string)
+	}
+	for k, v := range labels {
+		cm.Labels[k] = v
+	}
+	if cm.Annotations == nil {
+		cm.Annotations = make(map[string]string)
+	}
+	for k, v := range annotations {
+		cm.Annotations[k] = v
+	}
+	_, err := clientset.CoreV1().ConfigMaps(namespace).Update(context.TODO(), cm, v1.UpdateOptions{})
```

Comment:
Here you should use the correct `context`, Maybe the one coming from the http request or the one for k8s